bots/doc.go

```diff
@@ -1,69 +1,135 @@
+an integrated CONIKS client. The CONIKS client software generates and
+stores a key pair for each of the user's registered accounts.
+bots assumes that users authorize CONIKS clients to connect to their
+third-party accounts as part of the communication client's authorization.
+Much like these communication services, bots assumes that obtaining a user's
+credentials needed for the authorization is a sufficiently large burden
+to deter most adversaries from attempting an impersonation attack.
+
+Account proxies: The main component of bots are trusted proxy
+servers, each of which connect to a reserved account with a designated
+third-party service using an authorization protocol such as OAuth.
+This allows CONIKS clients and the proxies to communicate directly using a
+service's communication protocol.
+
+A proxy only accepts and relays CONIKS registration requests for accounts
+with its designated third-party service, and the requests must be sent by
+an authorized CONIKS client via the service's protocol to the proxy's
+reserved account. As a result, a proxy trusts that any requests sent
 by CONIKS clients to its account originated from uncompromised
 third-party accounts.
 
-To further restrict the allowable sources of registration requests, the
-CONIKS server only accepts incoming registration requests originating
-from the account proxy, and trusts that the proxy has
-performed the necessary third-party account verifications.
+CONIKS servers: bots proxies relay verified CONIKS registration requests
+to a designated CONIKS server. For authentication, proxies digitally
+sign all requests they relay. The server then only accepts incoming
+registration requests originating from authenticated account proxies.
+
+Privacy
+
+bots seeks to provide the same strong privacy guarantees as CONIKS.
+CONIKS clients using bots only store the generated key pairs and
+registered third-party usernames. bots only sends the username and
+the public key as an opaque data blob to the corresponding account proxy.
+bots also ensures that the clients do not gain access to any other
+third-party account information.
+
+The account proxies only have access to a user's username with their
+designated service and the received public key, and this is
+the only user information they send to the server.
+Proxies cannot access any other account information.
 
 Third-Party Account Verification Protocol
 
-1. The user adds her username to the CONIKS client, and connects to the corresponding account with the third-party service.
+1. The user registers her third-party account with her communication
+client, thereby authorizing the client and the integrated CONIKS client
+to connect to her third-party account.
+
+2. The CONIKS client generates a key pair for the new user account
+and stores the public and private key locally.
+
+3. On behalf of the connected account, the client sends a registration
+request
+    req = (username, opaque public key data blob)
```

Comment:
I think we can remove the `username` part, and let the bot insert this field into the request (as my current implementation).